root/static/scripts/common/components/ArtistCreditLink.js

```diff
@@ -10,8 +10,7 @@ const EntityLink = require('./EntityLink');
 const ArtistCreditLink = ({artistCredit, showDeleted = true, ...props}) => {
   const names = artistCredit.names;
   let parts = [];
-  for (let i = 0; i < names.size; i++) {
```

Comment:
`.size` should never be `undefined` if you're passing in an `Immutable.List` object, which is what `artistCreditFromArray` creates, so you should use that and revert this.

(Aside 1: `.map` creates a new list and then throws it away here; you wanted `forEach`, since you only wanted to iterate the list for side effects, not create a new list from the old one.)

(Aside 2: Immutable.js is being removed in https://github.com/metabrainz/musicbrainz-server/pull/606, so in the future we won't have to worry about converting plain arrays to Immutable lists.)